java/src/test/java/actorworkflow/CustomerLoyaltyTest.java

```diff
@@ -0,0 +1,176 @@
+package actorworkflow;
+
+import io.temporal.api.enums.v1.WorkflowIdReusePolicy;
+import io.temporal.client.WorkflowClient;
+import io.temporal.client.WorkflowExecutionAlreadyStarted;
+import io.temporal.client.WorkflowOptions;
+import io.temporal.testing.TestWorkflowRule;
+import io.temporal.testing.WorkflowReplayer;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.io.File;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Objects;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+/**
+ * Unit test for {@link CustomerLoyaltyWorkflow}.
+ */
+public class CustomerLoyaltyTest {
+
+    @Rule
+    public final TestWorkflowRule testWorkflowRule =
+            TestWorkflowRule.newBuilder()
+                    .setWorkflowTypes(CustomerLoyaltyWorkflowImpl.class)
+                    .setDoNotStart(true)
+                    .build();
+
+    @Test
+    public void testAddPoints() {
+        CustomerLoyaltyActivities activities = mock(CustomerLoyaltyActivities.class);
+        testWorkflowRule.getWorker().registerActivitiesImplementations(activities);
+        testWorkflowRule.getTestEnvironment().start();
+
+        // Get a workflow stub using the same task queue the worker uses.
+        WorkflowOptions workflowOptions =
+                WorkflowOptions.newBuilder()
+                        .setTaskQueue(testWorkflowRule.getTaskQueue())
+                        .setWorkflowIdReusePolicy(
+                                WorkflowIdReusePolicy.WORKFLOW_ID_REUSE_POLICY_REJECT_DUPLICATE)
+                        .build();
+        CustomerLoyaltyWorkflow workflow =
+                testWorkflowRule
+                        .getWorkflowClient()
+                        .newWorkflowStub(CustomerLoyaltyWorkflow.class, workflowOptions);
+
+        // Start workflow asynchronously to not use another thread to signal.
+        WorkflowClient.start(workflow::customerLoyalty, new Customer("123"));
+
+        StatusTier targetStatus = StatusTier.STATUS_TIERS.get(1);
+        workflow.addLoyaltyPoints(targetStatus.minimumPoints());
+        StatusTier customerStatus = workflow.getStatus();
+        assertEquals(customerStatus, targetStatus);
+
+        workflow.cancelAccount();
+        testWorkflowRule.getTestEnvironment().sleep(Duration.ofSeconds(1));
```

Comment:
i think test never waits for execution to complete
coudl do something like WorkflowStub.fromTyped(workflow).getResult(.....) and test that it completes without exception or something

Reply:
added that here and to the other tests, too.